src/main/java/com/minecolonies/core/client/gui/AbstractBuildingWindow.java

```diff
@@ -75,66 +51,47 @@ public AbstractBuildingWindow(final BOWindow parent, final B buildingView, final
     {
         super(parent, resource);
         this.buildingView = buildingView;
-        this.iconRandom = new Random(buildingView.getID().hashCode());
 
-        int nextTabIndex = 0;
+        final TabsWindowCapability tabsWindowCapability = registerModule(TabsWindowCapability::new, new Random(buildingView.getID().hashCode()));
```

Comment:
Still called capability.
Also I don't think we should need to register this on each constructor call.
Also, we don't need a new random for each